### PR TITLE
Log the publish method used when publishing

### LINKED ISSUE #26283: Log the WebPublishMethod
### Is your feature request related to a problem? Please describe.

We should log PublishProtocol so that we can query what publish mechanisms are actually used by customers, similar to https://github.com/dotnet/sdk/issues/25928. This will help us verify which methods users are actually using and direct focus accordingly.

### PR DESCRIPTION
**Pull request by baronfel**
Closes #26283

This adds the `PublishProtocol` to the set of data gathered about a publish. The intent is to allow determining the usage of different publish types, like Folder, MSDeploy, and eventually Containers.  This property is typically set from the `WebPublishMethod` that many Publish Profiles set, and it's used to pick a set of protocol-specific targets/props to import to actually _perform_ the related publish operation.

### CODE REVIEW

**Review by dsplaisted — APPROVED** [review]
Make sure to update the documentation on telemetry data points that we collect.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets [hunk 1]
```diff
@@ -72,7 +72,7 @@ Copyright (c) .NET Foundation. All rights reserved.
          the published assets were copied. -->
     <Message Importance="High" Text="$(MSBuildProjectName) -> $([System.IO.Path]::GetFullPath('$(PublishDir)'))" />
 
-    <AllowEmptyTelemetry EventName="PublishProperties" EventData="PublishReadyToRun=$(PublishReadyToRun);PublishTrimmed=$(PublishTrimmed);PublishSingleFile=$(PublishSingleFile);PublishAot=$(PublishAot)" />
+    <AllowEmptyTelemetry EventName="PublishProperties" EventData="PublishReadyToRun=$(PublishReadyToRun);PublishTrimmed=$(PublishTrimmed);PublishSingleFile=$(PublishSingleFile);PublishAot=$(PublishAot)PublishProtocol=$(PublishProtocol);" />
```
Comment (AntonLapounov) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets:75]:
```suggestion
    <AllowEmptyTelemetry EventName="PublishProperties" EventData="PublishReadyToRun=$(PublishReadyToRun);PublishTrimmed=$(PublishTrimmed);PublishSingleFile=$(PublishSingleFile);PublishAot=$(PublishAot);PublishProtocol=$(PublishProtocol)" />
```

**Review by AntonLapounov — CHANGES_REQUESTED** [review]
Please move the semicolon to the correct position.

**Review by AntonLapounov — APPROVED** [review]

**Review by rainersigwald — APPROVED** [review]

**Review by joeloff — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets (hunk 1 above)
Comment (baronfel) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets]: Thanks for this catch, @AntonLapounov - I messed this up when I cherry-picked my initial commit from the release/6.0.4xx branch to main to re-target it.

**Review by skundlony — APPROVED** [review]

**Pull request merged**